back/app/models/idea.rb

```diff
@@ -183,25 +183,10 @@ def custom_form
   end
 
   def input_term
```

Comment:
@kogre I did a bit of refactoring here while trying to understand what was happening. The code remains largely the same, but also filtering the phases on `supports_input_term?`. I was planning to use `supports_input_term?` in other places, but in the end perhaps this predicate could be just removed. There is a very extensive test coverage on this, so we can be quite confident that behaviour is still preserved. However, I wonder if we would want to filter out proposal phases in certain situations. Probably not though, as proposal phases wouldn't be linked to the idea, unless they are also the creation phase (in which case we always take the input term of the creation phase).

Reply:
If found another use for `supports_input_term?`: in the validations and callbacks (setting the default value)